libs/src/services/discoveryProvider/DiscoveryProviderSelection.test.js

```diff
@@ -0,0 +1,363 @@
+const nock = require('nock')
+const assert = require('assert')
+const semver = require('semver')
+
+const DiscoveryProviderSelection = require('./DiscoveryProviderSelection')
+
+const mockEthContracts = (urls, currrentVersion) => ({
+  getCurrentVersion: async () => currrentVersion,
+  getNumberOfVersions: async (spType) => 2,
+  getVersion: async (spType, queryIndex) => {
+    return ['1.2.2', '1.2.3'][queryIndex]
+  },
+  getServiceProviderList: async () => urls,
+  isValidSPVersion: (version1, version2) => {
+    return (
+      semver.major(version1) === semver.major(version2) &&
+      semver.minor(version1) === semver.minor(version2) &&
+      semver.patch(version2) >= semver.patch(version1)
+    )
+  }
+})
+
+describe('DiscoveryProviderSelection', () => {
+  beforeEach(() => {
+    const LocalStorage = require('node-localstorage').LocalStorage
+    const localStorage = new LocalStorage('./local-storage')
+    localStorage.removeItem('@audius/libs:discovery-provider-timestamp')
+  })
+
+  it('selects a healthy service', async () => {
+    const healthy = 'https://healthy.audius.co'
+    nock(healthy)
+      .get('/health_check')
+      .reply(200, {
+        service: 'discovery-provider',
+        version: '1.2.3',
+        block_difference: 0
+      })
+
+    const s = new DiscoveryProviderSelection(
+      {},
+      mockEthContracts([healthy], '1.2.3')
+    )
+    const service = await s.select()
+    assert.strictEqual(service, healthy)
+  })
+
+  it('selects a healthy service with an unhealthy one present', async () => {
+    const healthy = 'https://healthy.audius.co'
+    nock(healthy)
+      .get('/health_check')
+      .reply(200, {
+        service: 'discovery-provider',
+        version: '1.2.3',
+        block_difference: 0
+      })
+    const unhealthy = 'https://unhealthy.audius.co'
+    nock(unhealthy)
+      .get('/health_check')
+      .reply(400, {
+        service: 'discovery-provider',
+        version: '1.2.3',
+        block_difference: 0
+      })
+
+    const s = new DiscoveryProviderSelection(
+      {},
+      mockEthContracts([healthy, unhealthy], '1.2.3')
+    )
+    const service = await s.select()
+    assert.strictEqual(service, healthy)
+  })
+
+  it('prefers the correct vesion', async () => {
+    const healthy = 'https://healthy.audius.co'
+    nock(healthy)
+      .get('/health_check')
+      .reply(200, {
+        service: 'discovery-provider',
+        version: '1.2.3',
+        block_difference: 0
+      })
+    const outdated = 'https://outdated.audius.co'
+    nock(outdated)
+      .get('/health_check')
+      .reply(200, {
+        service: 'discovery-provider',
+        version: '1.2.2',
+        block_difference: 0
+      })
+
+    const s = new DiscoveryProviderSelection(
+      {},
+      mockEthContracts([healthy, outdated], '1.2.3')
```

Comment:
is there ever a case where `getVersion()` returns an old version and in that case the dp logic selects the older version? 

like
```
    const s = new DiscoveryProviderSelection(
      {},
      mockEthContracts([healthy, outdated], '1.2.2')
     )

     const service = await s.select() // will be the 1.2.2 service
```

Reply:
yes! there are a few tests that check this. see
`can select an old version`